### PR TITLE
ME-2300 Policy tester in cli

### PR DESCRIPTION
**Pull request by atoonk**
# Description

Policy tester option in CLI 
```
$ ./border0-cli policy test --name orgwide --email andree@border0.com --ip 206.214.246.9
Testing Policy: orgwide 8ebbcd53-bea4-4d8c-b3d9-80d35cfe2e99
✅ Policy Passed!

policy: "orgwide", Email address matched policy (andree@border0.com)
policy: "orgwide", IP address matched policy (206.214.246.96)
policy: "orgwide", Country code matched policy (CA)
policy: "orgwide", Date window matched policy (2023-11-15T06:36:21Z)
policy: "orgwide", Time window matched policy (06:36)

The following actions would be allowed:
{
  "http": [
    "*"
  ],
  "ssh": [
    "*"
  ],
  "tls": [
    "*"
  ]
}
```

```
$ ./border0-cli policy test --name orgwide --email andree@bla.xom --ip 1.2.3.4
Testing Policy: orgwide 8ebbcd53-bea4-4d8c-b3d9-80d35cfe2e99
⛔ Policy Failed

policy: "orgwide", Email address didn't match policy (andree@bla.xom)
policy: "orgwide", Country code didn't match policy (AU)
```

Fixes # (issue)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: cmd/policy.go [hunk 1]
```diff
@@ -259,6 +259,74 @@ func policyShow(cmd *cobra.Command, args []string) {
...
+			fmt.Println("✅ Policy Passed!")
+			fmt.Printf("\n")
+
+			// now print the reasons
+			for _, reason := range body.Info.Allowed {
+				fmt.Println(reason)
+			}
+			fmt.Println("\nThe following actions would be allowed:")
+			// pretty print the actions
+			jsonData, _ := json.MarshalIndent(body.Actions, "", "  ")
```
Comment (adrianosela) [inline cmd/policy.go:317]: Errcheck

File: cmd/policy.go [hunk 2]
```diff
@@ -259,6 +259,74 @@ func policyShow(cmd *cobra.Command, args []string) {
...
+			// This is for the colored JSON output
+			var colorData map[string]interface{}
+			json.Unmarshal([]byte(jsonData), &colorData)
+			f := colorjson.NewFormatter()
+			f.Indent = 2
+			colorString, _ := f.Marshal(colorData)
```
Comment (adrianosela) [inline cmd/policy.go:323]: Errcheck

File: internal/api/models/policy.go [hunk 3]
```diff
@@ -26,6 +26,20 @@ type Policy struct {
...
 }
 
+type PolicyTest struct {
+	Email     string `json:"email" binding:"required"`
+	IPAddress string `json:"ip_address" binding:"required"`
+	Time      string `json:"time" binding:"required"`
+}
+
+type PolicyTestRespone struct {
+	Actions map[string][]string `json:"Actions,omitempty"`
```
Comment (adrianosela) [inline internal/api/models/policy.go:36]: The api returns capitalized  json keys?

Comment (atoonk) [reply internal/api/models/policy.go:36]: yes it appears so
```
$ curl -s -d @testp.json  -H "x-access-token: $(cat ~/.border0/token)" https://api.border0.com/api/v1/policy/8ebbcd53-bea4-4d8c-b3d9-8xxxx/test | jq .
{
  "Actions": {
    "http": [
      "*"
    ],
    "ssh": [
      "*"
```

**Review by adrianosela — APPROVED** [review]

**Pull request merged**